src/pages/index.js

```diff
@@ -25,37 +26,115 @@ editProfileValidator.enableValidation();
 const addCardValidator = new FormValidator(config, newCardEditForm);
 addCardValidator.enableValidation();
 
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
+
+const removeCardValidator = new FormValidator(config, removeCardForm);
+removeCardValidator.enableValidation();
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "eeca6a67-c0e1-4af7-ac48-d61d2e5d5520",
+    "Content-Type": "application/json",
+  },
+});
+
+let cardSection;
+
+api.getInitialCards().then((cards) => {
+  cardSection = new Section(
+    {
+      items: cards,
+      renderer: (cardData) => {
+        const cardElement = getCardElement(cardData);
+        cardSection.addItem(cardElement);
+      },
+    },
+    ".cards__list"
+  );
+  cardSection.renderItems();
+});
+const userInfo = new UserInfo({
+  userName: "#profile-title-js",
+  userAbout: "#profile-description-js",
+  userAvatar: ".profile__image",
+});
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo(userData);
+});
+
 /**================================================================================================
  *                                         FUNCTION
  *================================================================================================**/
 
 function getCardElement(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleCardDelete,
+    handleLikeClick
+  );
   return card.getView();
 }
 /**------------------------------------------------------------------------------------------------
  *                                         EVENT HANDLERS
  *------------------------------------------------------------------------------------------------**/
 
-function handleProfileFormSubmit(formData) {
-  userInfo.setUserInfo({ name: formData.title, job: formData.description });
-  profileEditPopup.close();
-}
-
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileFormSubmit
+  (formData) => {
+    profileEditPopup.renderLoading(true);
+    api.setUserInfo(formData).then((userData) => {
+      userInfo.setUserInfo(userData);
+      profileEditPopup.renderLoading(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.